system-tests/README.md

```diff
@@ -1,62 +1,106 @@
+MVD System tests can be executed locally against a local MVD instance. MVD runs three EDC Connectors and one Registration Service.
 
-From the build result, download the artifact named `testing-configuration` and extract the file `.env` into the `system-tests` directory (note that the file could be hidden in your file explorer due to its prefix).
+First please make sure that you are able to build MVD locally as described in [Building MVD project](#building-mvd-project) section.
 
-In the file, add the application client secret value under the `APP_CLIENT_SECRET` key. It is used to access Key Vault.
+- We need to build EDC Connector launcher and Registration Service launcher.
+- Go to MVD root folder. And execute
 
-Build the EDC launcher:
+    ```bash
+    ./gradlew -DuseFsVault="true" :launcher:shadowJar
+    ```
 
-```
-./gradlew :launcher:shadowJar
-```
+- Go to Registration service root folder. And execute
 
-Run EDC consumer, provider and data seeding:
+    ```bash
+    ./gradlew :launcher:shadowJar
+    ```
 
-```
-docker-compose -f system-tests/docker-compose.yml up --build
-```
+- Update Registration service launcher path in `system-tests/docker-compose.yml` file. Look for section `#UPDATE_REGISTRATION_SERVICE_LAUNCHER_PATH_HERE#` and update it e.g. `/home/user/RegistrationService/launcher`.
 
-In the commands below, adapt the variable values marked with `$` to use the value from the `.env` file.
+- Start MVD using docker-compose.yml file.
 
-Login in to Azure:
-```
-az login --service-principal --user "$APP_CLIENT_ID" --password "$APP_CLIENT_SECRET" --tenant "$APP_TENANT_ID"
-```
+    ```bash
+    docker-compose -f system-tests/docker-compose.yml up --build
+    ```
 
-| ℹ️ Information                                                |
-| :----------------------------------------------------------- |
-| You could also login interactively with your user identity (`az login`), and [grant yourself at least the *Key Vault Secrets User*](https://docs.microsoft.com/azure/key-vault/general/rbac-guide) role to the Key Vault below. A good option is to grant the *Key Vault Secrets Officer* at the subscription level to the whole development team, so they can read and write secrets on MVD deployments as needed. |
+- This will start three EDC Connectors, one Registration Service, one HTTP Nginx Server to serve DIDs, Azurite blob storage service and also will seed initial required data using a [postman collection](../deployment/data/MVD.postman_collection.json).
 
-Run tests:
+- `newman` docker container will automatically stop after seeding initial data from postman scripts.
 
-```
-CONSUMER_EU_KEY_VAULT="$CONSUMER_EU_KEY_VAULT" CONSUMER_US_KEY_VAULT="$CONSUMER_US_KEY_VAULT" ./gradlew :system-tests:test
-```
+- EDC Connectors needs to be registered using Registration Service CLI client jar. After publishing RegistrationService locally the client jar should be available under `RegistrationService-Root/client-cli/build/libs` folder.
```

Comment:
The step-by-step instructions could be replaced by small scripts that automate most of it. Have you considered this? I think we could have:
- `startMVD.local..sh`: builds dependencies and starts MVD
-  `runTests.local.sh`: registers participants and runs tests

Reply:
@marcgs Yes could be. But how we will maintain & validate such script ? Because we are not using it in CI. Only script used here is to register participant and that is also used in CI to cover it on workflow runs.

Reply:
I would use the same scripts to run the tests against local MVD on CI. IMO the main advantage would be to get rid of the many steps for running MVD locally which can lead to errors, but it is not a must have either.

Reply:
As discussed offline leaving it as it as in this PR. As it can be addressed later. Thanks